frontend/components/side_panels/HostSidePanel/HostSidePanel.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useState, useCallback } from "react";
+import { filter, remove } from "lodash";
+
+import Button from "components/buttons/Button";
+// @ts-ignore
+import InputField from "components/forms/fields/InputField";
+// @ts-ignore
+import PanelGroup from "components/side_panels/HostSidePanel/PanelGroup";
+// @ts-ignore
+import SecondarySidePanelContainer from "components/side_panels/SecondarySidePanelContainer";
+import { ILabel } from "interfaces/label";
+import { PLATFORM_LABEL_DISPLAY_ORDER } from "utilities/constants";
+
+import PlusIcon from "../../../../assets/images/icon-plus-16x16@2x.png";
+
+const baseClass = "host-side-panel";
+
+interface IHostSidePanelProps {
+  labels: ILabel[];
+  onAddLabelClick: (evt: React.MouseEvent<HTMLButtonElement>) => void;
+  onLabelClick: (selectedLabel: ILabel) => boolean;
+  selectedFilter: string;
+  canAddNewLabel: boolean;
+}
+
+const HostSidePanel = (props: IHostSidePanelProps): JSX.Element => {
+  const {
+    labels,
+    onAddLabelClick,
+    onLabelClick,
+    selectedFilter,
+    canAddNewLabel,
+  } = props;
+
+  const [labelFilter, setLabelFilter] = useState("");
+
+  const onFilterLabels = useCallback(
+    (filterString: string): void => {
+      setLabelFilter(filterString.toLowerCase());
+    },
+    [setLabelFilter]
+  );
+
+  const allHostLabels = filter(labels, { type: "all" });
+
+  const hostPlatformLabels = (() => {
+    const unorderedList: ILabel[] = labels.filter(
+      (label) => label.type === "platform"
+    );
+
+    let orderedList: ILabel[] = [];
+    PLATFORM_LABEL_DISPLAY_ORDER.forEach((name) => {
+      orderedList.push(
+        ...remove(unorderedList, (label) => label.name === name)
+      );
+    });
+    orderedList = orderedList.concat(unorderedList);
```

Comment:
What's the goal here?

Reply:
@martavis 
So when I wrote it, my goal here was take the rest of the unordered list (whatever was left after the forEach does its thing) and tack it onto the end of the ordered list. At the time, I was thinking it might be good to avoid a similar bug in the future. But happy to dial it back in the interest of simplicity now that I moved us towards enumerating the platforms in the constants file.